ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala

```diff
@@ -1170,7 +1170,15 @@ private class JdbcLedgerDao(
     ~ str("observers").? map flatten)
 
   private val SQL_SELECT_CONTRACT =
-    SQL(queries.SQL_SELECT_CONTRACT)
+    SQL("""select cd.contract
+        |from contract_data cd
+        |left join contracts c on cd.id=c.id
+        |left join contract_witnesses cowi on cowi.contract_id = c.id and witness = {party}
+        |left join contract_divulgences codi on codi.contract_id = cd.id and party = {party}
+        |where
+        |  cd.id={contract_id} and
+        |  c.archive_offset is null and
+        |  (cowi.witness is not null or codi.party is not null)""".stripMargin)
```

Comment:
Maybe keep the newline before the closing quotes?

Reply:
Addressed by https://github.com/digital-asset/daml/pull/4545/commits/890ac2f1fe52e97f66cc6cbf5939b804706c9ab6